time stamp
float64 distance
float64 angle